### ISSUE TITLE
Feature: Implement Tuned Lens for improved layer-wise predictions

### ISSUE THREAD

**Issue by prof-schacht** [coairesearch/mlxterp#2]
## Summary

Implement the **Tuned Lens** technique from [Belrose et al. (2023)](https://arxiv.org/abs/2303.08112) as an extension to the existing `logit_lens()` functionality. The tuned lens trains a small affine transformation for each layer to correct for coordinate system mismatches, producing more accurate intermediate predictions.

## Motivation

The current `logit_lens()` implementation directly projects intermediate hidden states through the final layer norm and unembedding matrix. This assumes all layers share the same coordinate system, which is often not true - representations can be rotated, shifted, or stretched between layers.

The tuned lens addresses this by learning an affine "translator" (Wx + b) for each layer that maps hidden states to the final layer's coordinate system before unembedding. This results in:
- More accurate intermediate predictions
- More reliable interpretability insights
- Better detection of when predictions "crystallize"

#### 1. TunedLens Module (`mlxterp/tuned_lens.py`)

```python
class TunedLens(nn.Module):
    """Learned affine translators for each layer."""
    
    def __init__(self, num_layers: int, hidden_dim: int):
        super().__init__()
        self.translators = [nn.Linear(hidden_dim, hidden_dim) for _ in range(num_layers)]
        # Initialize close to identity
        for t in self.translators:
            t.weight = mx.eye(hidden_dim)
            t.bias = mx.zeros((hidden_dim,))
    
    def __call__(self, h: mx.array, layer_idx: int) -> mx.array:
        return self.translators[layer_idx](h)
    
    def save(self, path: str): ...
    def load(cls, path: str) -> 'TunedLens': ...
```

#### 2. Training Function

```python
def train_tuned_lens(
    model: InterpretableModel,
    dataset: List[str],
    num_steps: int = 250,
    learning_rate: float = 1.0,
    chunk_size: int = 2048,
    save_path: Optional[str] = None
) -> TunedLens:
    """Train tuned lens translators using KL divergence loss."""
```

**Training Details (from paper):**
- Optimizer: SGD with Nesterov momentum (0.9)
- Learning rate: 1.0 with linear decay over 250 steps
- Gradient clipping: norm 1.0
- Loss: KL divergence between translator prediction and final output
- Data: Chunks of length 2048 from validation/training set

#### 3. Integration with AnalysisMixin

Add to `mlxterp/analysis.py`:

```python
def tuned_logit_lens(
    self,
    text: str,
    tuned_lens: TunedLens,
    top_k: int = 1,
    layers: Optional[List[int]] = None,
    plot: bool = False
) -> Dict[int, List[List[tuple]]]:
    """Apply tuned lens for improved layer-wise predictions."""
```

### Implementation Steps

- [ ] **Step 1: Create TunedLens class** (mlxterp/tuned_lens.py)
  - Implement `TunedLens` module with affine translators
  - Add identity initialization for stability
  - Implement `save()` and `load()` methods using safetensors

- [ ] **Step 2: Implement training function**
  - Create `train_tuned_lens()` function
  - Implement KL divergence loss computation
  - Use SGD with Nesterov momentum and linear LR decay
  - Add gradient clipping
  - Support chunk-based training for long sequences

- [ ] **Step 3: Integrate with InterpretableModel**
  - Add `train_tuned_lens()` method to model (like `train_sae()`)
  - Add `load_tuned_lens()` method
  - Add `tuned_logit_lens()` method to AnalysisMixin

- [ ] **Step 4: Add visualization support**
  - Extend existing logit lens heatmap to support tuned lens
  - Add comparison view (raw vs tuned)

- [ ] **Step 5: Write tests**
  - Test TunedLens initialization and forward pass
  - Test training loop convergence
  - Test save/load functionality
  - Test integration with InterpretableModel

- [ ] **Step 6: Documentation and examples**
  - Add API documentation
  - Create tutorial notebook (examples/tutorials/01b_tuned_lens/)
  - Update README with tuned lens example

## API Design

```python
from mlxterp import InterpretableModel

model = InterpretableModel("mlx-community/Llama-3.2-1B-Instruct")

# Option 1: Train new tuned lens
tuned_lens = model.train_tuned_lens(
    dataset=texts,
    num_steps=250,
    save_path="tuned_lens_llama3.2.mlx"
)

# Option 2: Load pre-trained
tuned_lens = model.load_tuned_lens("tuned_lens_llama3.2.mlx")

# Use tuned lens
results = model.tuned_logit_lens(
    "The capital of France is",
    tuned_lens=tuned_lens,
    layers=[0, 5, 10, 15],
    plot=True
)

# Compare with raw logit lens
raw_results = model.logit_lens("The capital of France is", layers=[0, 5, 10, 15])
```

## File Changes

| File | Change |
|------|--------|
| `mlxterp/tuned_lens.py` | New file - TunedLens class and training |
| `mlxterp/analysis.py` | Add `tuned_logit_lens()` method |
| `mlxterp/model.py` | Add `train_tuned_lens()`, `load_tuned_lens()` methods |
| `mlxterp/__init__.py` | Export TunedLens |
| `tests/test_tuned_lens.py` | New file - tests |
| `examples/tutorials/01b_tuned_lens/` | New directory - tutorial |

## References

- **Paper:** [Eliciting Latent Predictions from Transformers with the Tuned Lens](https://arxiv.org/abs/2303.08112)
- **Original Implementation:** https://github.com/AlignmentResearch/tuned-lens
- **Documentation:** https://tuned-lens.readthedocs.io/

## Labels

`enhancement`, `interpretability`, `planned`
